### PR TITLE
chore: Improve release process with release-it

### LINKED ISSUE #414: Add changelog generation in release script
Based on #16 in this ticket, we should integrate it into our release script.

### PR DESCRIPTION
**Pull request by marlass**
Closes #414

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,16 @@
     "test:core:lib": "concurrently \"ng test core --code-coverage\" \"ng test storefrontlib --code-coverage\"",
     "test:storefront:lib": "ng test storefrontlib --sourceMap --code-coverage",
     "build:ssr": "ng run storefrontapp:server && webpack --config projects/ssr/webpack.server.config.js --progress --colors",
-    "start:ssr": "node dist/ssr/server"
+    "start:ssr": "node dist/ssr/server",
+    "release:styles": "cd projects/storefrontstyles && release-it && cd ../..",
+    "release:lib": "cd projects/storefrontlib && release-it && cd ../..",
+    "release:core": "cd projects/core && release-it && cd ../..",
+    "release-with-changelog:styles": "cd projects/storefrontstyles && release-it --config .release-it.changelog.json && cd ../..",
```
Comment (hackergil) [inline package.json]: Maybe we can rename this to release:styles:with-changelog? Same for the others

File: projects/core/.release-it.changelog.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "git": {
+    "requireCleanWorkingDir": true,
+    "requireUpstream": false,
+    "tagName": "core-${version}",
+    "commitMessage": "Release core-${version}",
```
Comment (hackergil) [inline projects/core/.release-it.changelog.json]: We use 'Bumping version to core-${version}' as a commit message

File: projects/core/.release-it.changelog.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    "tagAnnotation": "Release core-${version}"
+  },
+  "npm": {
+    "publish": true,
+    "publishPath": "./../../dist/core",
+    "tag": "next"
```
Comment (hackergil) [inline projects/core/.release-it.changelog.json]: shouldn't tag be 'latest'?

Comment (marlass) [reply projects/core/.release-it.changelog.json]: Changed

File: projects/core/.release-it.changelog.json (hunk 2 above)
Comment (marlass) [reply projects/core/.release-it.changelog.json]: Changed to `Bumping core to version ${version}` according to previous tags and commits

**Comment by marlass** [comment]
1. Docs proposal (don't know where to put them. In repo or in internal wiki?):
``` markdown
# Releasing packages

After testing release branch and packages locally with fresh app instance we can bump package version, create git tag, publish package on npm and create a release in github. To make it easier there is an npm script for each package that takes care of all those steps.

Example:
- we decide to release core and styles library
- test release branch and packages
- release core package with command `npm run release:core:with-changelog`
- release styles package with command `npm run release:styles:with-changelog`
- merge release branch to master and then develop

**Warning:** Use npm instead of yarn to run release script. There might be some errors with yarn on npm publish step.

**Info:** For publishing release in github you need to set GITHUB_TOKEN environment variable in your terminal (`export GITHUB_TOKEN="your_token_goes_here"`). If you forgot to do that script will fail at the beginning and inform you about it.

Running release script will prompt you with series of questions:
- select version bump
- create commit?
- create tag?
- push to remote repository?
- publish to npm? (there can be a followup question for one time password)
- create a release in github?

In normal flow you would answer `yes` to all yes/no questions.
Release each package with changelog (team agreed to do this).

`release-it` can be run with flag `--dry-run` that will only output commands that would be normally executed. You can modify npm scripts to test this.

To find out more about all `release-it` options go to [package github repository](https://github.com/webpro/release-it)
```

2. Answer in docs proposal

**Pull request merged**